streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSource.java

```diff
@@ -92,33 +97,36 @@ public void init(final org.apache.kafka.streams.processor.ProcessorContext conte
         }
 
         @Override
-        public void process(final K key, final V value) {
+        public void process(final Record<KIn, VIn> record) {
             // if the key is null, then ignore the record
-            if (key == null) {
-                LOG.warn(
+            if (record.key() == null) {
+                context.recordMetadata().ifPresent(recordMetadata -> LOG.warn(
```

Comment:
Nice touch! Since we only need the metadata for the context in the log, what do you think about logging `"Skipping record due to null key. Topic, partition, and offset not known."` if the metadata is absent?

Reply:
Agreed. I added some more context to the messages and make them consistent across the class. Let me know it looks.